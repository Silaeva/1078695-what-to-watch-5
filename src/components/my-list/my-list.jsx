import React, {useEffect} from "react";
import PropTypes from "prop-types";
import FilmCardList from "../film-card-list/film-card-list";
import LogoHeader from "../logo-header/logo-header";
import UserBlock from "../user-block/user-block";
import PageFooter from "../page-footer/page-footer";
import {connect} from "react-redux";
import {fetchFavoriteFilms} from "../../store/api-actions";
import filmProp from "../../film-prop";

const MyList = (props) => {
  const {favoriteFilms, onFilmCardClick, getFavoriteFilms, isFavoritesLoading, isFavoritesLoadError} = props;

  useEffect(() => {
    getFavoriteFilms();
  }, []);

  const getMyListPageContent = () => {
    if (isFavoritesLoading && !isFavoritesLoadError) {
      return (
        <p>Loading...</p>
      );
    } else if (isFavoritesLoadError) {
      return (
        <p>Sorry, something went wrong!</p>
      );
    }

    return (
      <FilmCardList films={favoriteFilms} onFilmCardClick={onFilmCardClick} />
    );
  };

  return (
    <div className="user-page">
      <header className="page-header user-page__head">

        <LogoHeader />

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock />

      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>

        {getMyListPageContent()}

      </section>

      <PageFooter />

    </div>
  );
};

MyList.propTypes = {
  favoriteFilms: PropTypes.arrayOf(filmProp),
  onFilmCardClick: PropTypes.func.isRequired,
  isFavoritesLoading: PropTypes.bool.isRequired,
  isFavoritesLoadError: PropTypes.bool.isRequired,
  getFavoriteFilms: PropTypes.func.isRequired
};

const mapStateToProps = ({DATA}) => ({
  favoriteFilms: DATA.favoriteFilms,
  isFavoritesLoading: DATA.isFavoritesLoading,
  isFavoritesLoadError: DATA.isFavoritesLoadError
});

const mapDispatchToProps = (dispatch) => ({
  getFavoriteFilms() {
    dispatch(fetchFavoriteFilms());
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
