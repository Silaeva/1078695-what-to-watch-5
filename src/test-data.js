const films = [
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 1,
    title: `Pulp Fiction`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    genre: `Crime`,
    year: 1994,
    duration: 153,
    director: `Quentin Tarantino`,
    actors: `John Travolta, Uma Thurman, Samuel L. Jackson`,
    rating: {
      score: 1.5,
      count: 163242
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    bgColor: `#795433`,
    isFavorite: false
  },
  {
    srcVideo: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 2,
    title: `Snatch`,
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Comedy`,
    year: 2000,
    duration: 104,
    director: `Guy Ritchie`,
    actors: `Jason Statham, Brad Pitt, Benicio Del Toro`,
    rating: {
      score: 5,
      count: 54654654
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg`,
    bgColor: `#FDFDFC`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    id: 3,
    title: `War of the Worlds`,
    description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Adventure`,
    year: 2005,
    duration: 116,
    director: `Steven Spielberg`,
    actors: `Tom Cruise, Dakota Fanning, Tim Robbins`,
    rating: {
      score: 9,
      count: 23425
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg`,
    bgColor: `#9B7E61`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 4,
    title: `Pulp Fiction`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    genre: `Crime`,
    year: 1994,
    duration: 153,
    director: `Quentin Tarantino`,
    actors: `John Travolta, Uma Thurman, Samuel L. Jackson`,
    rating: {
      score: 1.5,
      count: 163242
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    bgColor: `#795433`,
    isFavorite: false
  },
  {
    srcVideo: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 5,
    title: `Snatch`,
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Comedy`,
    year: 2000,
    duration: 104,
    director: `Guy Ritchie`,
    actors: `Jason Statham, Brad Pitt, Benicio Del Toro`,
    rating: {
      score: 5,
      count: 54654654
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg`,
    bgColor: `#FDFDFC`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    id: 6,
    title: `War of the Worlds`,
    description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Adventure`,
    year: 2005,
    duration: 116,
    director: `Steven Spielberg`,
    actors: `Tom Cruise, Dakota Fanning, Tim Robbins`,
    rating: {
      score: 9,
      count: 23425
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg`,
    bgColor: `#9B7E61`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 7,
    title: `Pulp Fiction`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    genre: `Crime`,
    year: 1994,
    duration: 153,
    director: `Quentin Tarantino`,
    actors: `John Travolta, Uma Thurman, Samuel L. Jackson`,
    rating: {
      score: 1.5,
      count: 163242
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    bgColor: `#795433`,
    isFavorite: false
  },
  {
    srcVideo: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 8,
    title: `Snatch`,
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Comedy`,
    year: 2000,
    duration: 104,
    director: `Guy Ritchie`,
    actors: `Jason Statham, Brad Pitt, Benicio Del Toro`,
    rating: {
      score: 5,
      count: 54654654
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg`,
    bgColor: `#FDFDFC`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    id: 9,
    title: `War of the Worlds`,
    description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Adventure`,
    year: 2005,
    duration: 116,
    director: `Steven Spielberg`,
    actors: `Tom Cruise, Dakota Fanning, Tim Robbins`,
    rating: {
      score: 9,
      count: 23425
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg`,
    bgColor: `#9B7E61`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 10,
    title: `Pulp Fiction`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    genre: `Crime`,
    year: 1994,
    duration: 153,
    director: `Quentin Tarantino`,
    actors: `John Travolta, Uma Thurman, Samuel L. Jackson`,
    rating: {
      score: 1.5,
      count: 163242
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    bgColor: `#795433`,
    isFavorite: false
  },
  {
    srcVideo: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    id: 11,
    title: `Snatch`,
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Comedy`,
    year: 2000,
    duration: 104,
    director: `Guy Ritchie`,
    actors: `Jason Statham, Brad Pitt, Benicio Del Toro`,
    rating: {
      score: 5,
      count: 54654654
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg`,
    bgColor: `#FDFDFC`,
    isFavorite: true
  },
  {
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    id: 12,
    title: `War of the Worlds`,
    description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
    image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    genre: `Adventure`,
    year: 2005,
    duration: 116,
    director: `Steven Spielberg`,
    actors: `Tom Cruise, Dakota Fanning, Tim Robbins`,
    rating: {
      score: 9,
      count: 23425
    },
    bgImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg`,
    bgColor: `#9B7E61`,
    isFavorite: true
  }
];

const comments = [
  {
    id: 1,
    text: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    rating: 4,
    user: `Christina`,
    userId: 19,
    date: `2020-10-09T13:38:44.769Z`
  }, {
    id: 2,
    text: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    rating: 9,
    user: `Sophie`,
    userId: 25,
    date: `2020-11-04T13:38:44.769Z`
  }, {
    id: 1,
    text: `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
    rating: 7,
    user: `Jack`,
    userId: 11,
    date: `2020-10-09T13:38:44.769Z`
  }
];

const noop = () => {};

const genre = `Comedy`;

export {films, comments, noop, genre};
