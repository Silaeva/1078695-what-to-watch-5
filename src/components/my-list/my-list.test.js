import React from "react";
import renderer from "react-test-renderer";
import {MyList} from "./my-list";
import {films, noop} from "../../test-data";

jest.mock(`../film-card-list/film-card-list`, () => `FilmCardList`);
jest.mock(`../logo-header/logo-header`, () => `LogoHeader`);
jest.mock(`../user-block/user-block`, () => `UserBlock`);
jest.mock(`../page-footer/page-footer`, () => `PageFooter`);

describe(`Should MyList render correctly`, () => {
  it(`When load success`, () => {
    const tree = renderer
    .create(
        <MyList
          favoriteFilms={films}
          onFilmCardClick={noop}
          getFavoriteFilms={noop}
          isFavoritesLoading={false}
          isFavoritesLoadError={false}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When films is loading`, () => {
    const tree = renderer
    .create(
        <MyList
          favoriteFilms={films}
          onFilmCardClick={noop}
          getFavoriteFilms={noop}
          isFavoritesLoading={true}
          isFavoritesLoadError={false}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When films load error`, () => {
    const tree = renderer
    .create(
        <MyList
          favoriteFilms={films}
          onFilmCardClick={noop}
          getFavoriteFilms={noop}
          isFavoritesLoading={false}
          isFavoritesLoadError={true}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
