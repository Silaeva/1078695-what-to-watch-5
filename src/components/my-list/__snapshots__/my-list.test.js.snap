// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MyList render correctly When films is loading 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <LogoHeader />
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <UserBlock />
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <p>
      Loading...
    </p>
  </section>
  <PageFooter />
</div>
`;

exports[`Should MyList render correctly When films load error 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <LogoHeader />
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <UserBlock />
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <p>
      Sorry, something went wrong!
    </p>
  </section>
  <PageFooter />
</div>
`;

exports[`Should MyList render correctly When load success 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <LogoHeader />
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <UserBlock />
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <FilmCardList
      films={
        Array [
          Object {
            "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
            "bgColor": "#795433",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
            "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
            "director": "Quentin Tarantino",
            "duration": 153,
            "genre": "Crime",
            "id": 1,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
            "isFavorite": false,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 163242,
              "score": 1.5,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "Pulp Fiction",
            "year": 1994,
          },
          Object {
            "actors": "Jason Statham, Brad Pitt, Benicio Del Toro",
            "bgColor": "#FDFDFC",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg",
            "description": "Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond",
            "director": "Guy Ritchie",
            "duration": 104,
            "genre": "Comedy",
            "id": 2,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 54654654,
              "score": 5,
            },
            "srcVideo": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            "title": "Snatch",
            "year": 2000,
          },
          Object {
            "actors": "Tom Cruise, Dakota Fanning, Tim Robbins",
            "bgColor": "#9B7E61",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg",
            "description": "As Earth is invaded by alien tripod fighting machines, one family fights for survival.",
            "director": "Steven Spielberg",
            "duration": 116,
            "genre": "Adventure",
            "id": 3,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            "rating": Object {
              "count": 23425,
              "score": 9,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "War of the Worlds",
            "year": 2005,
          },
          Object {
            "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
            "bgColor": "#795433",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
            "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
            "director": "Quentin Tarantino",
            "duration": 153,
            "genre": "Crime",
            "id": 4,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
            "isFavorite": false,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 163242,
              "score": 1.5,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "Pulp Fiction",
            "year": 1994,
          },
          Object {
            "actors": "Jason Statham, Brad Pitt, Benicio Del Toro",
            "bgColor": "#FDFDFC",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg",
            "description": "Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond",
            "director": "Guy Ritchie",
            "duration": 104,
            "genre": "Comedy",
            "id": 5,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 54654654,
              "score": 5,
            },
            "srcVideo": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            "title": "Snatch",
            "year": 2000,
          },
          Object {
            "actors": "Tom Cruise, Dakota Fanning, Tim Robbins",
            "bgColor": "#9B7E61",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg",
            "description": "As Earth is invaded by alien tripod fighting machines, one family fights for survival.",
            "director": "Steven Spielberg",
            "duration": 116,
            "genre": "Adventure",
            "id": 6,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            "rating": Object {
              "count": 23425,
              "score": 9,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "War of the Worlds",
            "year": 2005,
          },
          Object {
            "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
            "bgColor": "#795433",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
            "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
            "director": "Quentin Tarantino",
            "duration": 153,
            "genre": "Crime",
            "id": 7,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
            "isFavorite": false,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 163242,
              "score": 1.5,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "Pulp Fiction",
            "year": 1994,
          },
          Object {
            "actors": "Jason Statham, Brad Pitt, Benicio Del Toro",
            "bgColor": "#FDFDFC",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg",
            "description": "Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond",
            "director": "Guy Ritchie",
            "duration": 104,
            "genre": "Comedy",
            "id": 8,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 54654654,
              "score": 5,
            },
            "srcVideo": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            "title": "Snatch",
            "year": 2000,
          },
          Object {
            "actors": "Tom Cruise, Dakota Fanning, Tim Robbins",
            "bgColor": "#9B7E61",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg",
            "description": "As Earth is invaded by alien tripod fighting machines, one family fights for survival.",
            "director": "Steven Spielberg",
            "duration": 116,
            "genre": "Adventure",
            "id": 9,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            "rating": Object {
              "count": 23425,
              "score": 9,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "War of the Worlds",
            "year": 2005,
          },
          Object {
            "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
            "bgColor": "#795433",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
            "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
            "director": "Quentin Tarantino",
            "duration": 153,
            "genre": "Crime",
            "id": 10,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
            "isFavorite": false,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 163242,
              "score": 1.5,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "Pulp Fiction",
            "year": 1994,
          },
          Object {
            "actors": "Jason Statham, Brad Pitt, Benicio Del Toro",
            "bgColor": "#FDFDFC",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg",
            "description": "Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond",
            "director": "Guy Ritchie",
            "duration": 104,
            "genre": "Comedy",
            "id": 11,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "rating": Object {
              "count": 54654654,
              "score": 5,
            },
            "srcVideo": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            "title": "Snatch",
            "year": 2000,
          },
          Object {
            "actors": "Tom Cruise, Dakota Fanning, Tim Robbins",
            "bgColor": "#9B7E61",
            "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg",
            "description": "As Earth is invaded by alien tripod fighting machines, one family fights for survival.",
            "director": "Steven Spielberg",
            "duration": 116,
            "genre": "Adventure",
            "id": 12,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg",
            "isFavorite": true,
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg",
            "previewVideo": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            "rating": Object {
              "count": 23425,
              "score": 9,
            },
            "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            "title": "War of the Worlds",
            "year": 2005,
          },
        ]
      }
      onFilmCardClick={[Function]}
    />
  </section>
  <PageFooter />
</div>
`;
